src/algebra/ordered_monoid.lean

```diff
@@ -49,6 +49,29 @@ class ordered_add_comm_monoid (α : Type*) extends add_comm_monoid α, partial_o
 
 attribute [to_additive] ordered_comm_monoid
 
+/-- A linearly ordered commutative monoid with a zero element. -/
+class linear_ordered_comm_monoid_with_zero (α : Type*)
+  extends linear_order α, comm_monoid_with_zero α, ordered_comm_monoid α :=
+(zero_le_one : (0:α) ≤ 1)
+(lt_of_mul_lt_mul_left := λ x y z, by {
+  -- type-class inference uses a meaningless `linear_order` without this!
```

Comment:
But how can it even find that meaningless instance? I'm a bit confused...

Reply:
Try it and see?

Reply:
I've clarified the comment a bit and golfed the proof, although the trick is still here.